allauth/account/forms.py

```diff
@@ -223,17 +233,31 @@ def __init__(self, *args, **kwargs):
         super(BaseSignupForm, self).__init__(*args, **kwargs)
         # field order may contain additional fields from our base class,
         # so take proper care when reordering...
+        self.email_field_name = 'email'
         field_order = ['email', 'username']
+        if app_settings.SIGNUP_EMAIL_VERIFICATION:
+            del self.fields["email"]
```

Comment:
> Is there a reason to swap out email fields?

> If someone wanted to switch the SIGNUP_EMAIL_VERIFICATION from True to False or False to True, keeping the first field name consistent would result in a better user experience for anyone who had the form open during the setting swap.

It was made for consistency with password fields - instead of having `email` and `email_confirmation` fields it was implemented fields `email1` and `email2`. Having different approaches in email and password verification might cause confusion. 

Original `email` was kept as is by default - for backward compatibility, since email verification is not widely used and would be optional.